server/server.go

```diff
@@ -489,6 +492,66 @@ func New(
 	return s, nil
 }
 
+func (s *Server) wipeIfAppropriate(ctx context.Context) {
+	if !s.config.InternalWipe {
+		return
+	}
+	// First, check if the node was rebooted.
+	// We know this happened because the VersionFile (which lives in a tmpfs)
+	// will not be there.
+	shouldWipeContainers, err := version.ShouldCrioWipe(s.config.VersionFile)
+	if err != nil {
+		log.Infof(ctx, "checking whether cri-o should wipe containers: %v", err)
+	}
+
+	// there are two locations we check before wiping:
+	// one in a temporary directory. This is to check whether the node has rebooted.
+	// if so, we should remove containers
+	// another is needed in a persistent directory. This is to check whether we've upgraded
+	// if we've upgraded, we should wipe images
+	shouldWipeImages, err := version.ShouldCrioWipe(s.config.VersionFilePersist)
+	if err != nil {
+		log.Infof(ctx, "%v: triggering wipe of images", err.Error())
+	}
+
+	shouldWipeContainers = shouldWipeContainers || shouldWipeImages
+
+	// First, save the images we should be wiping
+	// We won't remember if we wipe all the containers first
+	imagesToWipe := make([]string, 0)
+	if shouldWipeImages {
+		containers, err := s.ContainerServer.ListContainers()
+		if err != nil {
+			logrus.Warnf("Failed to list containers: %v", err)
+		}
+		for _, c := range containers {
+			imagesToWipe = append(imagesToWipe, c.ImageRef())
+		}
+	}
+
+	if shouldWipeContainers {
+		for _, sb := range s.ContainerServer.ListSandboxes() {
+			cleanupFunc := func() error {
+				if err := s.stopPodSandbox(ctx, sb); err != nil {
+					return err
+				}
+				return s.removePodSandbox(ctx, sb)
+			}
+			if err := cleanupFunc(); err != nil {
```

Comment:
What is the point of using the cleanupFunc here? It looks like it's equivalent to this (simpler) code:

```go
for _, sb := range s.ContainerServer.ListSandboxes() {
   err := s.stopPodSandbox(ctx, sb)
   if err == nil {
      err = s.removePodSandbox(ctx, sb)
   }
   if err != nil {
      log.Warnf(ctx, "Failed to cleanup pod %s: %v", sb.ID(), err)
   }
}
```

Reply:
because we expect this to fail in some cases (if the node has restarted but the networking plugin pods haven't been created, then any non-host network pods can't have their networking resources cleaned up)

Reply:
oh I see, we don't yet actually do that. I think it was just to reduce churn between commits, but I can drop it for this commit and update in a later one